lib/attache/upload.rb

```diff
@@ -25,7 +25,14 @@ def _call(env, config)
           return [500, config.headers_with_cors.merge('X-Exception' => 'Local file failed'), []]
         end
 
-        config.async(:storage_create, relpath: relpath, cachekey: cachekey) if config.storage && config.bucket
+        if config.storage && config.bucket
+          request.body.rewind if request.body.respond_to?(:rewind)
+          if Attache.outbox.write(request_hostname(env), relpath, request.body)
+            config.async(:storage_create, relpath: relpath, cachekey: cachekey)
```

Comment:
Will the ocal file be removed when `config.async` succeeds or is that the concern of something else?